pkg/performanceprofile/controller/performanceprofile/hypershift/components/handler.go

```diff
@@ -0,0 +1,280 @@
+		klog.Infof("Kubelet Config %q exists under the namespace %q", name, operatorNamespace)
+		return true
+	}
+
+	if _, err := resources.GetRuntimeClass(ctx, h.dataPlaneClient, name); !k8serrors.IsNotFound(err) {
+		klog.Infof("Runtime class %q exists under the hosted cluster", name)
+		return true
+	}
+
+	if _, err := resources.GetMachineConfig(ctx, h.controlPlaneClient, machineconfig.GetMachineConfigName(profileName)); !k8serrors.IsNotFound(err) {
+		klog.Infof("Machine Config %q exists under the namespace %q", name, operatorNamespace)
+		return true
+	}
+	return false
+}
+
+func (h *handler) Apply(ctx context.Context, obj client.Object, recorder record.EventRecorder, options *components.Options) error {
+	// TODO  handle naming
+	// Jose left the following comment:
+	// Make PerformanceProfile names unique if a PerformanceProfile is duplicated across NodePools
+	// for example, if one ConfigMap is referenced in multiple NodePools
+	// need to check whether this comment is relevant or not.
+	instance, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		return fmt.Errorf("wrong type conversion; want=ConfigMap got=%T", obj)
+	}
+
+	s, ok := instance.Data[hypershift.TuningKey]
+	if !ok {
+		return fmt.Errorf("key named %q not found in ConfigMap %q", hypershift.TuningKey, client.ObjectKeyFromObject(obj).String())
+	}
+
+	profile := &performancev2.PerformanceProfile{}
+	if err := hypershift.DecodeManifest([]byte(s), h.scheme, profile); err != nil {
+		return err
+	}
+
+	if profileutil.IsPaused(profile) {
+		klog.Infof("ignoring reconcile loop for pause performance profile %s", profile.Name)
+		return nil
+	}
+
+	ctrRuntime, err := h.getContainerRuntimeName(ctx, profile)
+	if err != nil {
+		return fmt.Errorf("could not determine high-performance runtime class container-runtime for profile %q; %w", profile.Name, err)
+	}
+	klog.Infof("using %q as high-performance runtime class container-runtime for profile %q", ctrRuntime, profile.Name)
+	options.MachineConfig.DefaultRuntime = ctrRuntime
+
+	mfs, err := manifestset.GetNewComponents(profile, options)
+	if err != nil {
+		return err
+	}
+
+	cm, err := h.encapsulateObjInConfigMap(instance, mfs.Tuned, profile.Name, tunedConfigMapConfigKey, tunedConfigMapLabel)
+	if err != nil {
+		return err
+	}
+	err = createOrUpdateTunedConfigMap(ctx, h.controlPlaneClient, cm)
+	if err != nil {
+		return err
+	}
+
+	cm, err = h.encapsulateObjInConfigMap(instance, mfs.MachineConfig, profile.Name, mcoConfigMapConfigKey, ntoGeneratedMachineConfigLabel)
+	if err != nil {
+		return err
+	}
+	err = createOrUpdateMachineConfigConfigMap(ctx, h.controlPlaneClient, cm)
+	if err != nil {
+		return err
+	}
+
+	cm, err = h.encapsulateObjInConfigMap(instance, mfs.KubeletConfig, profile.Name, mcoConfigMapConfigKey, ntoGeneratedMachineConfigLabel)
+	if err != nil {
+		return err
+	}
+	err = createOrUpdateKubeletConfigConfigConfigMap(ctx, h.controlPlaneClient, cm)
```

Comment:
There is one redundant `config` in the function name

Reply:
Thanks